packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -120,7 +106,21 @@ contract L2Genesis is Deployer {
     ///         Sets the precompiles, proxies, and the implementation accounts to be `vm.dumpState`
     ///         to generate a L2 genesis alloc.
     function runWithStateDump() public {
-        runWithOptions(OutputMode.DEFAULT_LATEST, artifactDependencies());
+        Fork fork = Config.fork();
+        if (fork == Fork.NONE) {
+            // Will revert if no deploy config can be found either.
+            fork = latestDeployConfigGenesisFork();
+            console.log("L2Genesis: using deploy config fork: %d", uint256(fork));
+        } else {
+            console.log("L2Genesis: using env var fork: %d", uint256(fork));
+        }
+        runWithOptions(Config.outputMode(), fork, artifactDependencies());
+    }
+
+    function latestDeployConfigGenesisFork() internal returns (Fork) {
+        DeployConfig cfg = DeployConfig(address(uint160(uint256(keccak256(abi.encode("optimism.deployconfig"))))));
```

Comment:
Is there a reason as to why we cannot use the globally defined deploy config?

Reply:
I totally should use that globally defined deploy config 😅 I wasn't aware it's there as I'm a total forge script noob 😬

Reply:
All good!